applications/StructuralMechanicsApplication/custom_response_functions/adjoint_schemes/adjoint_structural_static_scheme.h

```diff
@@ -0,0 +1,413 @@
+
+    typedef typename BaseType::TSystemVectorType SystemVectorType;
+
+    typedef typename BaseType::LocalSystemVectorType LocalSystemVectorType;
+
+    typedef typename BaseType::LocalSystemMatrixType LocalSystemMatrixType;
+
+    typedef typename BaseType::DofsArrayType DofsArrayType;
+
+    ///@}
+    ///@name Life Cycle
+    ///@{
+
+    /// Constructor.
+    AdjointStructuralStaticScheme(Parameters& rParameters, AdjointStructuralResponseFunction::Pointer pResponseFunction)
+        : Scheme<TSparseSpace, TDenseSpace>()
+    {
+        KRATOS_TRY;
+
+        Parameters default_params(R"(
+        {
+            "scheme_type": "adjoint_structural"
+        })");
+
+        rParameters.ValidateAndAssignDefaults(default_params);
+
+        mpResponseFunction = pResponseFunction;
+
+        KRATOS_CATCH("");
+    }
+
+    /// Destructor.
+    ~AdjointStructuralStaticScheme() override
+    {
+    }
+
+    ///@}
+    ///@name Operators
+    ///@{
+
+    ///@}
+    ///@name Operations
+    ///@{
+
+    void Initialize(ModelPart& rModelPart) override
+    {
+        KRATOS_TRY;
+
+        BaseType::Initialize(rModelPart);
+
+        // Allocate auxiliary memory
+        int num_threads = OpenMPUtils::GetNumThreads();
+        mAdjointValues.resize(num_threads);
+
+        mpResponseFunction->Initialize();
+
+        KRATOS_CATCH("");
+    }
+
+    void InitializeSolutionStep(ModelPart& rModelPart,
+                                SystemMatrixType& rA,
+                                SystemVectorType& rDx,
+                                SystemVectorType& rb) override
+    {
+        KRATOS_TRY;
+
+        // Sensitivities are generally computed as a time integral. For steady
+        // problems, we set the time step to -1.0 (minus because adjoint is
+        // backward in time).
+        rModelPart.GetProcessInfo()[DELTA_TIME] = -1.0;
+
+        BaseType::InitializeSolutionStep(rModelPart, rA, rDx, rb);
+
+        // initialize the variables to zero.
+        #pragma omp parallel for
+        for (int k = 0; k< static_cast<int> (rModelPart.Nodes().size()); ++k) 
+        {
+            ModelPart::NodesContainerType::iterator it_node = rModelPart.NodesBegin()+ k;
+            noalias(it_node->FastGetSolutionStepValue(ADJOINT_DISPLACEMENT)) = ADJOINT_DISPLACEMENT.Zero();
+            noalias(it_node->FastGetSolutionStepValue(ADJOINT_ROTATION)) = ADJOINT_ROTATION.Zero();
+        }
+
+        mpResponseFunction->InitializeSolutionStep();
+
+        KRATOS_CATCH("");
+    }
+
+    void FinalizeSolutionStep(ModelPart& rModelPart,
+                              SystemMatrixType& rA,
+                              SystemVectorType& rDx,
+                              SystemVectorType& rb) override
+    {
+        KRATOS_TRY;
+
+        BaseType::FinalizeSolutionStep(rModelPart, rA, rDx, rb);
+
+        mpResponseFunction->FinalizeSolutionStep();
+
+        KRATOS_CATCH("");
+    }
+
+    /// Update adjoint.
+    void Update(ModelPart& rModelPart,
```

Comment:
I think this functionality was moved to the `DofUpdater` by @jcotela

Reply:
The newer schemes will be added in a separate branch. I would leave this as it is since we know it will be soon replaced.